apps/api/src/app/events/usecases/send-message/send-message-chat.usecase.ts

```diff
@@ -48,21 +59,39 @@ export class SendMessageChat extends SendMessageType {
       _environmentId: command.environmentId,
       _id: command.subscriberId,
     });
-    const content = await this.compileTemplate.execute(
-      CompileTemplateCommand.create({
-        templateId: 'custom',
-        customTemplate: chatChannel.template.content as string,
-        data: {
-          subscriber,
-          step: {
-            digest: !!command.events.length,
-            events: command.events,
-            total_count: command.events.length,
-          },
-          ...command.payload,
-        },
-      })
-    );
+    let content = '';
+    const data = {
+      subscriber,
+      step: {
+        digest: !!command.events.length,
+        events: command.events,
+        total_count: command.events.length,
+      },
+      ...command.payload,
+    };
+    try {
+      content = await this.compileTemplate.execute(
+        CompileTemplateCommand.create({
+          templateId: 'custom',
+          customTemplate: chatChannel.template.content as string,
+          data,
+        })
+      );
```

Comment:
If you move inside the try brace the lines from 96 to 102, you avoid to have to create the content variable outside of the brace and you avoid the return in the catch brace.

Reply:
Not sure why I would like to avoid that?

Reply:
For me the code becomes more readable, more imperative without having to jump to future lines. It took me a while to understand some of the bits. That's why I suggest it. Otherwise looks functionally fine!

Reply:
the think is that I do not like to catch stuff that is done below... Because then the message in the execution detail could be wrong, we could catch an error from something that is not related to creating the message content.

Reply:
I just understood what you were looking for. 👍🏻